app/src/main/java/com/amaze/filemanager/crashreport/ErrorActivity.java

```diff
@@ -174,74 +177,77 @@ private static String getStackTrace(final Throwable throwable) {
   @Override
   public void onCreate(final Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    setContentView(R.layout.activity_error);
-    final Toolbar toolbar = findViewById(R.id.toolbar);
-    setSupportActionBar(toolbar);
-
     final Intent intent = getIntent();
-
-    final ActionBar actionBar = getSupportActionBar();
-    if (actionBar != null) {
-      actionBar.setDisplayHomeAsUpEnabled(true);
-      actionBar.setTitle(R.string.error_report_title);
-      actionBar.setDisplayShowTitleEnabled(true);
-    }
-
-    final Button reportEmailButton = findViewById(R.id.errorReportEmailButton);
-    final Button reportTelegramButton = findViewById(R.id.errorReportTelegramButton);
-    final Button copyButton = findViewById(R.id.errorReportCopyButton);
-    final Button reportGithubButton = findViewById(R.id.errorReportGitHubButton);
-
-    userCommentBox = findViewById(R.id.errorCommentBox);
-    final TextView errorView = findViewById(R.id.errorView);
-    final TextView errorMessageView = findViewById(R.id.errorMessageView);
-
-    returnActivity = MainActivity.class;
-    errorInfo = intent.getParcelableExtra(ERROR_INFO);
-    errorList = intent.getStringArrayExtra(ERROR_LIST);
-
-    // important add guru meditation
-    addGuruMeditation();
-    currentTimeStamp = getCurrentTimeStamp();
-
-    reportEmailButton.setOnClickListener((View v) -> sendReportEmail());
-
-    reportTelegramButton.setOnClickListener(
-        (View v) -> {
-          FileUtils.copyToClipboard(this, buildMarkdown());
-          Toast.makeText(this, R.string.crash_report_copied, Toast.LENGTH_SHORT).show();
-          Utils.openTelegramURL(this);
-        });
-
-    copyButton.setOnClickListener(
-        (View v) -> {
-          FileUtils.copyToClipboard(this, buildMarkdown());
-          Toast.makeText(this, R.string.crash_report_copied, Toast.LENGTH_SHORT).show();
-        });
-
-    reportGithubButton.setOnClickListener(
-        (View v) -> {
-          FileUtils.copyToClipboard(this, buildMarkdown());
-          Toast.makeText(this, R.string.crash_report_copied, Toast.LENGTH_SHORT).show();
-          Utils.openURL(ERROR_GITHUB_ISSUE_URL, this);
-        });
-
-    // normal bugreport
-    buildInfo(errorInfo);
-    if (errorInfo.message != 0) {
-      errorMessageView.setText(errorInfo.message);
+    if (ErrorInfo.comparePackageInfo(intent.getPackage()) != 0) {
+      finish();
     } else {
```

Comment:
No else needed AFAIK.

Reply:
At least a `return` to get out of there would make me feel more safe ;)